### PR TITLE
switch to bluebird, cleanup promise chains, use bluebirds disposer pattern

### PR DESCRIPTION
**Pull request by phated**
Necessary reading https://github.com/petkaantonov/bluebird/blob/master/API.md#resource-management

### CODE REVIEW

File: lib/board.js [hunk 1]
```diff
@@ -1,138 +1,58 @@
...
-    }
-
-    function send(){
-      return bs2.bootload(serialPort, options.board, options.memory.data);
-    }
-
-    function close(){
-      return when.promise(function(resolve, reject) {
-
-        serialPort.on('error', function(err){
```
Comment (jacobrosenthal) [inline lib/board.js:88]: Close isn't async so this is necessary where the connection might be opened again immediately as it wont be successfully closed until one of these return.

Comment (phated) [inline lib/board.js:88]: write a test case because my way is the right way to do this stuff

File: lib/board.js [hunk 2]
```diff
@@ -1,138 +1,58 @@
...
 
-var com = require('serialport');
-var when = require('when');
-var nodefn = require('when/node');
-
+var SerialPort = require('serialport').SerialPort;
 var bs2 = require('bs2-programmer');
+var bluebird = require('bluebird');
 
-function Board(){
```
Comment (jacobrosenthal) [inline lib/board.js:9]: I agree currently theres no reason to have this be an object, but next week you'll probably want board.open and board.on('read') and board.write

Comment (phated) [inline lib/board.js:9]: don't add code until you use it

**Comment by phated** [comment]
I added the `serialport.closeAsync()` call we discussed today.

**Pull request merged**